google/cloud/security/scanner/scanners/bigquery_scanner.py

```diff
@@ -24,28 +28,70 @@
 class BigqueryScanner(base_scanner.BaseScanner):
     """Pipeline to pull Big Query acls data from DAO"""
 
-    def __init__(self, snapshot_timestamp):
+    def __init__(self, global_configs, scanner_configs, snapshot_timestamp,
+                 rules):
         """Initialization.
 
         Args:
             snapshot_timestamp (str): The snapshot timestamp
         """
         super(BigqueryScanner, self).__init__(
-            snapshot_timestamp)
-        self.snapshot_timestamp = snapshot_timestamp
+            global_configs,
+            scanner_configs,
+            snapshot_timestamp,
+            rules)
+        self.rules_engine = bigquery_rules_engine.BigqueryRulesEngine(
+            rules_file_path=self.rules,
+            snapshot_timestamp=self.snapshot_timestamp)
+        self.rules_engine.build_rule_book(self.global_configs)
+
+    def _flatten_violations(self, violations):
+        for violation in violations:
+            violation_data = {}
+            violation_data['dataset_id'] = violation.dataset_id
+            violation_data['access_domain'] = violation.domain
+            violation_data['access_user_by_email'] = violation.user_email
+            violation_data['access_special_group'] = violation.special_group
+            violation_data['access_group_by_email'] = violation.group_email
+            violation_data['role'] = violation.role
+            yield {
+                'resource_id': violation.resource_id,
+                'resource_type': violation.resource_type,
+                'rule_index': violation.rule_index,
+                'rule_name': violation.rule_name,
+                'violation_type': violation.violation_type,
+                'violation_data': violation_data
+            }
+
+    def _output_results(self, all_violations, resource_counts):
+        resource_name = 'violations'
```

Comment:
Would it make sense to move this resource type as a class variable? Something called violations_resource_name?